ocm_python_wrapper/cluster.py

```diff
@@ -262,26 +263,91 @@ def addon_info(self):
             self.addon_name
         ).to_dict()
 
-    def validate_addon_parameters(self, parameters):
+    def validate_and_update_addon_parameters(
+        self, user_parameters, use_api_defaults=True
+    ):
+        """Validate and update user input parameters against API's conditions and requirements.
+
+        Args:
+            user_parameters (List): User parameters
+                Example:
+                    parameters = [{"id": "has-external-resources", "value": "false"},
+                                    {"id": "aws-cluster-test-param", "value": "false"},]
+            use_api_defaults (bool): If true, set required parameter (which are not part of `user_parameters`) with
+                                            default value to not fail as missing.
+
+        Returns:
+            List: Updated parameters (if default values updated) to provide for installation.
```

Comment:
This function can return None is some cases.
Please make sure you always return list, empty one if needed.

Reply:
Done - we can only return `user_parameters` (list) which is set as an empty list in the base case.